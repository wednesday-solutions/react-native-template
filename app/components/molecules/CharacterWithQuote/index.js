import React from 'react';
import styled from 'styled-components/native';
import get from 'lodash/get';
import PropTypes from 'prop-types';
import { fonts } from '@themes';
import T from '@atoms/T';

const Result = styled(T)`
  ${fonts.style.standard()};
  text-align: center;
  margin-bottom: 5;
`;

const CharacterImage = styled.Image`
  height: 80px;
  width: 80px;
  margin: 0 auto;
`;

function CharacterWithQuote({ user }) {
  return (
    <>
      <Result
        id="wednesday_lover"
        values={{
          username: get(user, 'character', 'character')
        }}
      />
      <Result id="because" />
      <CharacterImage
        resizeMode="contain"
        testID="character-image"
        source={{ uri: get(user, 'image') }}
      />
      <Result text={get(user, 'quote')} />
    </>
  );
}

CharacterWithQuote.propTypes = {
  user: PropTypes.object
};

export default CharacterWithQuote;
