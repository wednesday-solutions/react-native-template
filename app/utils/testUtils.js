import React from 'react';
import { IntlProvider } from 'react-intl';
import { render } from '@testing-library/react-native';
import { Provider } from 'react-redux';
import createStore from 'app/rootReducer';
import { DEFAULT_LOCALE, translationMessages } from '@app/i18n';
import ConnectedLanguageProvider from '@atoms/LanguageProvider';

export const apiResponseGenerator = (ok, data) => ({
  ok,
  data
});
export const renderWithIntl = (children, renderFunction = render) =>
  renderFunction(
    <IntlProvider
      locale={DEFAULT_LOCALE}
      messages={translationMessages[DEFAULT_LOCALE]}
    >
      {children}
    </IntlProvider>
  );

export const renderProvider = children => {
  const { store } = createStore();
  return render(
    <Provider store={store}>
      <ConnectedLanguageProvider messages={translationMessages}>
        {children}
      </ConnectedLanguageProvider>
    </Provider>
  );
};

export const timeout = ms => new Promise(resolve => setTimeout(resolve, ms));
