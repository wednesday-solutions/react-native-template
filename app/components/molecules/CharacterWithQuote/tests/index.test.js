/**
 *
 * Tests for CharacterWithQuote
 *
 * @see https://github.com/react-boilerplate/react-boilerplate/tree/master/docs/testing
 *
 */

import React from 'react';
import get from 'lodash/get';
import { renderWithI18next } from '@utils/testUtils';
import CharacterWithQuote from '../index';

describe('<CharacterWithQuote />', () => {
  it('Should render and match the snapshot', () => {
    const baseElement = renderWithI18next(<CharacterWithQuote />);
    expect(baseElement).toMatchSnapshot();
  });

  it('Should render the Character name, image and quote provided as the user prop', () => {
    const props = {
      user: {
        character: 'Homer',
        image:
          'https://www.onthisday.com/images/people/homer-simpson-medium.jpg',
        quote: "D'Oh!"
      }
    };
    const { getByText, getByTestId } = renderWithI18next(
      <CharacterWithQuote {...props} />
    );
    expect(getByText('wednesday_lover')).toBeTruthy();
    expect(getByText(props.user.quote)).toBeTruthy();
    const characterImageURI = get(
      getByTestId('character-image'),
      '_fiber.pendingProps.source.uri'
    );
    expect(characterImageURI).toBe(props.user.image);
  });
});
