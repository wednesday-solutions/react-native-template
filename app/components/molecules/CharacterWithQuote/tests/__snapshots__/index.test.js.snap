// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CharacterWithQuote /> Should render and match the snapshot 1`] = `
Array [
  <Text
    style={
      Array [
        Object {
          "fontSize": 17,
          "fontWeight": "normal",
          "marginBottom": 5,
          "textAlign": "center",
        },
      ]
    }
  >
    character loves Wednesday
  </Text>,
  <Text
    style={
      Array [
        Object {
          "fontSize": 17,
          "fontWeight": "normal",
          "marginBottom": 5,
          "textAlign": "center",
        },
      ]
    }
  >
    because
  </Text>,
  <Image
    resizeMode="contain"
    source={
      Object {
        "uri": undefined,
      }
    }
    style={
      Array [
        Object {
          "height": 80,
          "marginBottom": 0,
          "marginLeft": "auto",
          "marginRight": "auto",
          "marginTop": 0,
          "width": 80,
        },
      ]
    }
    testID="character-image"
  />,
  <Text
    style={
      Array [
        Object {
          "fontSize": 17,
          "fontWeight": "normal",
          "marginBottom": 5,
          "textAlign": "center",
        },
      ]
    }
  />,
]
`;
